network/wsNetwork.go

```diff
@@ -843,15 +875,39 @@ func (wn *WebsocketNetwork) ServeHTTP(response http.ResponseWriter, request *htt
 		return
 	}
 
+	// check to see that the genesisID in the request URI is valid and matches the supported one.
```

Comment:
This section looks like it belongs in wn.checkHeaders().  These won't get tracked in networkConnectionsDroppedTotal metric.

Reply:
The networkConnectionsDroppedTotal metric is a good point. However, I disagree that it should be tested in the checkHeaders; it's being tested differently on the client and on the server, which is why it was removed from the checkHeaders function.